packages/relay-compiler/RelayCompilerPublic.js

```diff
@@ -15,11 +15,12 @@ const GraphQLCompilerContext = require('./graphql-compiler/core/GraphQLCompilerC
 const RelayCodeGenerator = require('./codegen/RelayCodeGenerator');
 const RelayFileWriter = require('./codegen/RelayFileWriter');
 const RelayIRTransforms = require('./core/RelayIRTransforms');
```

Comment:
Currently Relay only exports the predefined groups of transforms from https://github.com/facebook/relay/blob/a8af6d7dea1c087bd6c5ba1ca4a2f538d9989cff/packages/relay-compiler/core/RelayIRTransforms.js. However, other language plugins will want to re-use some of these just like the Flow generator does. We are currently doing that in our TS plugin by simply [plucking the ones that the Flow generator uses out from the pre-defined groups](https://github.com/kastermester/relay-compiler-language-typescript/blob/88854c6d57e98bede437c5c4534e46b6c6db7c10/src/TypeScriptGenerator.ts#L535-L539), but this is fragile and so a more stable solution would be appreciated.

What do you think would be the best way to do this? Export _all_ transformers or is it unlikely that a type generator would need any other builtin transforms than the ones that the Flow generator also uses and so we should just add a predefined group for type generators?

Reply:
I’m currently leaning towards creating a predefined group for type generators.